flocker/provision/_openstack.py

```diff
@@ -0,0 +1,154 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+OpenStack provisioner.
+"""
+
+from textwrap import dedent
+from time import time
+
+from effect.retry import retry
+from effect import Effect, Constant
+
+from ._libcloud import LibcloudProvisioner
+from ._install import (
+    provision,
+    task_install_ssh_key,
+    task_open_control_firewall,
+)
+from ._ssh import run_remotely
+
+from ._effect import sequence
+
+# XXX: Copied from _aws. Needs refactoring
+_usernames = {
+    'centos-7': 'centos',
+    'ubuntu-14.04': 'ubuntu',
+    'ubuntu-15.04': 'ubuntu',
+}
+
+
+def get_default_username(distribution):
+    """
+    Return the username available by default on a system.
+
+    :param str distribution: Name of the operating system distribution
+    :return str: The username made available by AWS for this distribution.
+    """
+    return _usernames[distribution]
+
+
+def provision_openstack(node, package_source, distribution, variants):
```

Comment:
Looking at the commit messages it may be different. In that case, probably every bit of it needs comments to explain why the bits are there (as contrast to all the other slightly different copies of this code.